numpyro/infer/reparam.py

```diff
@@ -100,11 +91,13 @@ def __call__(self, name, fn, obs):
                                      constraint=constraints.unit_interval)
         params["loc"] = fn.loc * centered
         params["scale"] = fn.scale ** centered
-        decentered_fn = type(fn)(**params).expand(batch_shape)
+        decentered_fn = type(fn)(**params)
+        decentered_fn = self._wrap(decentered_fn, event_dim)
+        decentered_fn = decentered_fn.expand(batch_shape)
```

Comment:
Agreed! We should correct event_dim first, then doing `expand`. However, this might raise an issue for the case:
`dist.Normal(0, 1).expand([10]).to_event(1)`, where `decentered_fn = dist.Normal(0, 1)` and it can't be expanded. I think we can do something like
```
decentered_fn = self._wrap(decentered_fn.expand(batch_shape + event_shape), event_dim)
```

Reply:
Thanks, it took me a while to understand. I just pushed an update to the `._wrap()` method which seems clean 😄

Reply:
Thanks for the fix! It looks great!